index.js

```diff
@@ -751,6 +826,30 @@
            keys.every(function(k) { return equals(self[k], other[k]); });
   }
 
+  function unionKeys(a, b) {
+    var keys = Object.keys(a);
+    var bKeys = Object.keys(b);
+    for (var i = 0; i < bKeys.length; i += 1) {
+      if (a[bKeys[i]] === void 0) keys.push(bKeys[i]);
```

Comment:
Let's follow the precedent set by `Object$prototype$map`, which uses an unguarded `for...in` loop. In other words, let's respect enumerable properties regardless of where they're "attached".